lib/improver/utilities/solar.py

```diff
@@ -0,0 +1,255 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+""" Utilites to do with day and night."""
+
+import numpy as np
+import math
+import datetime as dt
+import iris
+
+from improver.utilities.cube_checker import check_cube_coordinates
+from improver.utilities.temporal import iris_time_to_datetime
+from improver.utilities.spatial import (
+    lat_lon_determine, transform_grid_to_lat_lon)
+
+
+def solar_declination(day_of_year):
+    """
+    Calculate the Declination for the day of the year.
+
+    Args:
+        day_of_year (int):
+            Day of the year
+
+    Returns:
+        declination (float):
+            Declination in degrees.North-South
+    """
+    # Declination (degrees):
+    # = -(axial_tilt)*cos(360./orbital_year * day_of_year - solstice_offset)
+    declination = -23.5 * math.cos(math.radians(0.9856 * day_of_year + 9.3))
+    return declination
+
+
+def solar_hour_angle(longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar Hour angle for an array of longitudes.
+
+    Args:
+        longitudes (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0
+        day_of_year (int):
+            Day of the year
+        utc_hour (float):
+            Hour of the day in UTC
+
+    Returns:
+        solar_hour_angle (numpy.array)
+            Hour angle in degrees East-West
+    """
+    thetao = 2*math.pi*day_of_year/365.0
+    eqt = (0.000075 + 0.001868 * math.cos(thetao) -
+           0.032077 * math.sin(thetao) - 0.014615 * math.cos(2*thetao) -
+           0.040849 * math.sin(2*thetao))
+
+    # Longitudinal Correction from the Grenwich Meridian
+    lon_correction = 24.0*longitudes/360.0
+    # Solar time (hours):
+    solar_time = utc_hour + lon_correction + eqt*12/math.pi
+    # Hour angle (degrees):
+    solar_hour_angle = (solar_time - 12.0) * 15.0
+
+    return solar_hour_angle
+
+
+def solar_elevation(latitudes, longitudes, day_of_year, utc_hour):
+    """
+    Calculate the Solar elevation.
+
+    Args:
+        latitude (float or numpy.array):
+            A single Latitudes or array of Latitudes
+            longitudes needs to be between -90.0 and 90.0
+        longitude (float or numpy.array):
+            A single Longitude or array of Longitudes
+            longitudes needs to be between 180.0 and -180.0
```

Comment:
Correct use of singular and plural words. Ensure argument name is correctly used in the doc-string.

Reply:
Corrected